src/structures/guild_preference.ts

```diff
@@ -270,21 +325,25 @@ export default class GuildPreference {
      */
     async setGroups(groupIDs: MatchedArtist[]) {
         this.gameOptions.groups = groupIDs;
-        await this.updateGuildPreferences(true);
+        await this.updateGuildPreferences([{ name: GameOptionInternal.GROUPS, value: this.getGroups() }]);
     }
 
     /** Resets the groups option to the default value */
     async resetGroups() {
-        this.gameOptions.groups = null;
-        await this.updateGuildPreferences(true);
+        await this.setGroups(null);
     }
 
-    /** @returns the current selected groups, if the groups option is active */
+    /** @returns the current selected groups by ID, if the groups option is active */
     getGroupIDs(): number[] {
         if (this.gameOptions.groups === null) return [];
         return this.gameOptions.groups.map((x) => x.id);
     }
 
+    /** @returns the current selected groups (name and ID), if the groups option is active */
+    getGroups(): MatchedArtist[] {
```

Comment:
Why do we need this new function

Reply:
I decided to replace all direct accesses of `this.gameOptions` with the relevant getter methods, similar to how I replaced directly setting member variables with setters in the reset methods (for [less points of failure](https://github.com/Brainicism/KMQ_Discord/pull/754/files#diff-6a5f3a950d16f8e31e2693b67c9ade56a23da2ebec004a48d0261571653f17edL531)). However, there weren't any equivalent functions for groups, includes, or excludes, so I decided to add them for consistency. However, I concede that it was mainly just a style choice on my end, and I'll happily revert them if you want.

Reply:
Don't see a reason to expose this as a method